### PR TITLE
Segmentation on Actions (page url, page title...) should work for goals and ecommerce reports

### LINKED ISSUE #6593: PageURL Segmentation fails on the ecommerce page
**Problem:** If there is a segment selected (in our case "Page URL contains /ch/"), here will be no results on the Ecommerce Page (but there should be some results there). 
That problem exists in version 2.4.1, but also in the newest version (2.8.3). 
Maybe important: We use a cronjob to archive our reports.

Thanks for solving this problem.

### PR DESCRIPTION
**Pull request by mattab**
fixes #6593
- reproduced first the issue in a new `Goals.get` system test 
- then added integration test `test_getSelectQuery_whenJoinConversionOnAction_segmentUsesPageUrl()` to isolate the problem in the sql itself
- we no longer use `idlink_va` to JOIN log_conversion with log_visit or log_link_visit_action - instead we use `idvisit`
- it was reported & escalated by a [Piwik Enterprise](http://piwik.org/enterprise)  client. it is very good that we finally fix this bug, I will look again at `Bugs` in Short term to see if we missed any important ones such as this one.

### CODE REVIEW

File: tests/PHPUnit/System/expected/test_ecommerceOrderWithItems_SegmentVisitHasConvertedGoal__Goals.getItemsName_week.xml [hunk 1]
```diff
@@ -3,9 +3,9 @@
 	<row>
 		<label>PRODUCT name</label>
 		<nb_visits>1</nb_visits>
-		<nb_actions>1</nb_actions>
```
Comment (mattab) [inline tests/PHPUnit/System/expected/test_ecommerceOrderWithItems_SegmentVisitHasConvertedGoal__Goals.getItemsName_week.xml]: here we now have more data  (which is now correct)

File: tests/PHPUnit/System/EcommerceOrderWithItemsTest.php [hunk 2]
```diff
@@ -67,11 +67,24 @@ public function getApiForTesting()
                         'date' => $dateTime,
                         'periods' => array('day', 'week'),
                         'otherRequestParameters' => array('_leavePiwikCoreVariables' => 1),
-                        'segment' => 'pageUrl=@Another%20Product%20page',
+                        'segment' => 'pageUrl=@example.org%2Findex.htm',
```
Comment (mattab) [inline tests/PHPUnit/System/EcommerceOrderWithItemsTest.php]: here I fixed the system test, technically should have been in a separate PR as it's un-related

File: tests/PHPUnit/System/expected/test_ecommerceOrderWithItems_EcommerceOrderGoal_SegmentPageUrlContains__Goals.get_day.xml [hunk 3]
```diff
@@ -0,0 +1,33 @@
+<?xml version="1.0" encoding="utf-8" ?>
```
Comment (mattab) [inline tests/PHPUnit/System/expected/test_ecommerceOrderWithItems_EcommerceOrderGoal_SegmentPageUrlContains__Goals.get_day.xml]: these show `0` everywhere on master branch

**Comment by diosmosis** [comment]
looks good to me. primary key for log_conversion should be used, so should be ok that there's no index. merging.

**Pull request merged**